app/validators/placeholders/placeholder_validator.py

```diff
@@ -123,6 +136,13 @@ def validate_placeholder_transforms(self, transforms):
                 and argument_name == "answer_id"
             ):
                 self.validate_option_label_from_value_placeholder(argument)
+            if (
+                first_transform["transform"] in ["format_unit", "format_percentage"]
+                and argument_name == "value"
+            ):
+                self.validate_answer_type_correct(
+                    argument, first_transform["transform"]
+                )
```

Comment:
This needs to be done for all transforms. This is only validating the first one.

Reply:
Ok, added thanks!